src/app/SubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,160 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the classes corresponding to CHIP Interaction Model Event Generatorr Delegate.
+ *
+ */
+
+#pragma once
+
+#include <app/ReadClient.h>
+#include <lib/core/CHIPCore.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * A SubscriptionPersistenceDelegate is used to persist subscriptions when they are established.
+ *
+ * Allows application to append any type of TLV data as part of an event log entry. Events
+ * have a standard header applicable to all events and this class provides the
+ * ability to add additional data past such standard header.
+ */
+class SubscriptionResumptionStorage
+{
+public:
+    // TODO: Create RAII ScopedMemoryBuffer replacement container that does not require is_trivial elements
```

Comment:
Not needed. Overall, usage of iterator patterns such as in most other storage classes and "data providers" would have avoided the need of the publicly exposed API surface to require maintenance of large buffers in heap

Reply:
I've changed the storage API to use an iterator pattern as suggested.